### PR TITLE
Fix opening file dialogs in PySide

### LINKED ISSUE #5508: File > Open dialog doesn't work in pyside2
## 🐛 Bug

Trying to use the GUI to open a file in napari with pyside2 backend results in an error
First reported here: https://github.com/napari/napari/issues/5506#issuecomment-1403061389
because our conda bundles use pyside2.

```
AttributeError                            Traceback (most recent call last)
File ~/Dev/miniforge3/envs/nap-test/lib/python3.10/site-packages/napari/_qt/qt_viewer.py:735, in QtViewer._open_files_dialog(self=<napari._qt.qt_viewer.QtViewer(0x6000008e1810)>, choose_plugin=False)
    732 hist = get_open_history()
    733 dlg.setHistory(hist)
--> 735 filenames, _ = dlg.getOpenFileNames(
        dlg = <PySide2.QtWidgets.QFileDialog(0x600000851180) at 0x16bb672c0>
        self = <napari._qt.qt_viewer.QtViewer(0x6000008e1810) at 0x16fcf2140>
        trans = <napari.utils.translations.TranslationBundle object at 0x14f201930>
        hist = ['/Users/piotrsobolewski']
        hist[0] = '/Users/piotrsobolewski'
        QFileDialog = <class 'PySide2.QtWidgets.QFileDialog'>
        QFileDialog.DontUseNativeDialog = PySide2.QtWidgets.QFileDialog.Option.DontUseNativeDialog
    736     parent=self,
    737     caption=trans._('Select file(s)...'),
    738     directory=hist[0],
    739     options=(
    740         QFileDialog.DontUseNativeDialog
    741         if in_ipython()
    742         else QFileDialog.Options()
    743     ),
    744 )
    746 if (filenames != []) and (filenames is not None):
    747     for filename in filenames:

AttributeError: PySide2.QtWidgets.QFileDialog.getOpenFileNames(): unsupported keyword 'directory'
```
Looks like the bug being fixed here https://github.com/napari/napari/pull/5492/files for pyside6, so best to adjust that PR to also fix this.

## To Reproduce

1. install pyside2 backend (conda or pip)
2. `pip install napari` (not [all] so you don't get pyqt5)
3. run napari and try File > Open

## Expected behavior

The GUI menus should work!

## Environment

napari 0.4.17 (pip) arm64 macOS 13.1
pyside2 5.15.8 (conda)

### PR DESCRIPTION
**Pull request by Czaki**
# Description

Even if official documentation of qt says that the argument of `getOpenFileNames` is named `dir`, the `dir` is a builtin function in python, so PySide2, PyQt5, and PyQt6 use `directory` as the argument name. But PySide6 breaks this.   

The `PyQt6` does not support `QFileDialog.Options()` 

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

**Review by psobolewskiPhD — APPROVED** [review]
I merged this with the other pyside6 PR and it works nice.
Makes sense to me.

Out of curiosity, it's >10 lines duplicated—does it make sense from a maintenance PoV to make this a function?

**Comment by Czaki** [comment]
> Out of curiosity, it's >10 lines duplicated—does it make sense from a maintenance PoV to make this a function?

done

**Comment by psobolewskiPhD** [comment]
> > Out of curiosity, it's >10 lines duplicated—does it make sense from a maintenance PoV to make this a function?
> 
> done

Bah, you didn't have to! I was just asking! 😬 

BTW I get this message in console:
```
2023-01-19 21:08:39.248 napari[35804:10079388] +[CATransaction synchronize] called within transaction
```
Seems like it may be macOS 13 specific:
https://forum.qt.io/topic/141868/macos-catransaction-synchronize-called-within-transaction/7
Just putting this somewhere for future reference.

**Comment by Czaki** [comment]
> Bah, you didn't have to! I was just asking! grimacing

but it is improvement

File: napari/_qt/qt_viewer.py [hunk 1]
```diff
@@ -735,22 +736,32 @@ def _screenshot_dialog(self):
...
-                QFileDialog.DontUseNativeDialog
-                if in_ipython()
-                else QFileDialog.Options()
-            ),
-        )
+        open_kwargs = {
+            "parent": self,
+            "caption": caption,
+        }
+        if "PySide6" in QFileDialog.__module__:
```
Comment (goanpeca) [inline napari/_qt/qt_viewer.py]: Would it make sense to use `"pyside6" in QFileDialog.__module__.lower()`

Comment (psobolewskiPhD) [inline napari/_qt/qt_viewer.py]:
```suggestion
        if "PySide" in QFileDialog.__module__:
```

This bug also affects pyside2:
https://github.com/napari/napari/issues/5508

Comment (Czaki) [reply napari/_qt/qt_viewer.py]: done

Comment (Czaki) [reply napari/_qt/qt_viewer.py]: done

**Comment by psobolewskiPhD** [comment]
Kinda amazing this was missed this long, but I have to admit when I was new I used File > Sample and drag-n-drop and now I largely use those two plus programatic methods. So I'm never doing File > Open.

File: napari/_qt/qt_viewer.py (hunk 1 above)
Comment (psobolewskiPhD) [reply napari/_qt/qt_viewer.py]: works for me in pyside2.

**Comment by Czaki** [comment]
An badly it is not possible to test this in normal way.

Maybe I should bring issue to provide both PySide and PyQt bundles?

**Pull request merged**